lib/cadet/accounts/game_states.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Cadet.GameStates do
+  import Ecto.Repo
+
+  # currently in this module no error handling function
+  # has been implemented yet
+
+  def user_game_states(user) do
+    user.game_states
+  end
+
+  def user_collectibles(user) do
+    user.game_states["collectibles"]
+  end
+
+  def user_save_data(user) do
+    user.game_states["completed_quests"]
+  end
+
+  def update(user, new_game_states) do
+    if user.role == :student do
```

Comment:
```suggestion
  def update(user = %User{role: role}, new_game_states) do
    if role in @update_gamestate_roles do
```
Role checking in the Assessments module is usually done this way. This also adds an abstraction barrier to the roles.